src/Microsoft.TestPlatform.Client/DesignMode/DesignModeClient.cs

```diff
@@ -287,27 +287,34 @@ private void StartTestRun(TestRunRequestPayload testRunPayload, ITestRequestMana
             Task.Run(
             delegate
             {
+                var success = false;
+                var exception = default(Exception);
                 try
                 {
                     testRequestManager.ResetOptions();
 
                     var customLauncher = skipTestHostLaunch ?
                         DesignModeTestHostLauncherFactory.GetCustomHostLauncherForTestRun(this, testRunPayload) : null;
 
-                    testRequestManager.RunTests(testRunPayload, customLauncher, new DesignModeTestEventsRegistrar(this), this.protocolConfig);
+                    success = testRequestManager.RunTests(testRunPayload, customLauncher, new DesignModeTestEventsRegistrar(this), this.protocolConfig);
                 }
                 catch (Exception ex)
                 {
                     EqtTrace.Error("DesignModeClient: Exception in StartTestRun: " + ex);
+                    success = false;
+                    exception = ex;
+                }
 
+                if (!success)
+                {
                     // If there is an exception during test run request creation or some time during the process
                     // In such cases, TestPlatform will never send a TestRunComplete event and IDE need to be sent a run complete message
                     // We need recoverability in translationlayer-designmode scenarios
-                    var testMessagePayload = new TestMessagePayload { MessageLevel = TestMessageLevel.Error, Message = ex.ToString() };
+                    var testMessagePayload = new TestMessagePayload { MessageLevel = TestMessageLevel.Error, Message = exception?.ToString() ?? "Exception in StartTestRun." };
```

Comment:
Better error message? Also error  message need to be present in resource file. Should we take that much changes for 15.6?

Reply:
let's handle the exception case differently.
If we get a success false, we would have already sent the TestMessagePayload.